core/src/main/java/org/elasticsearch/bootstrap/ESPolicy.java

```diff
@@ -35,19 +36,20 @@
 
 /** custom policy for union of static and dynamic permissions */
 final class ESPolicy extends Policy {
-    
+
     /** template policy file, the one used in tests */
     static final String POLICY_RESOURCE = "security.policy";
     /** limited policy for scripts */
     static final String UNTRUSTED_RESOURCE = "untrusted.policy";
-    
+
     final Policy template;
     final Policy untrusted;
     final Policy system;
     final PermissionCollection dynamic;
     final Map<String,Policy> plugins;
 
     public ESPolicy(PermissionCollection dynamic, Map<String,Policy> plugins, boolean filterBadDefaults) {
+        dynamic.add(new LinkPermission("hard")); // nocommit - is this the right place to add this?
```

Comment:
yeah the link permission is a particularly bad one. it means you can access all files because of how permissions work on unix. I think its a good idea to contain it to just a particular piece of code rather than globally granted. In all cases it should be handled in the policy configuration file.

I think it can still work out though: personally, I don't think we should implement Store.java's copyFile with hard links on the regular basis anyway, instead I think its better implemented as a FilterDirectory that implements it this way solely for addIndexes (it seems to be a good tradeoff here to reduce transient space for that operation, but otherwise I think we should just copy bytes the simple way).